fedimint-core/src/db/mod.rs

```diff
@@ -1030,6 +1030,7 @@ pub struct DatabaseTransaction<'a> {
     tx: Box<dyn ISingleUseDatabaseTransaction<'a>>,
     decoders: ModuleDecoderRegistry,
     commit_tracker: CommitTracker,
+    on_commit_hooks: Vec<Box<maybe_add_send!(dyn FnOnce())>>,
```

Comment:
Would we want these commit hooks to return a `Result`?

Reply:
What would we do with the result? When the hooks get called we already committed the dbtx and if we were to run them before then they'd fire without knowing if the dbtx actually succeeded.